crates/uv-resolver/src/lock.rs

```diff
@@ -77,6 +78,53 @@ pub struct Lock {
 }
 
 impl Lock {
+    /// Deserialize the [`Lock`] from a TOML string.
+    pub fn from_toml(s: &str) -> Result<Lock, toml::de::Error> {
+        let mut lock: Lock = toml::from_str(s)?;
+
+        // Simplify all marker expressions based on the requires-python bound.
+        //
+        // This is necessary to ensure the a `Lock` deserialized from a lockfile compares
```

Comment:
What is the difference between fresh and from lock? We always know the current requires-python in the resolver (we determine it on fork construction), so marker construction should already have this information.

Also note that requires-python is the bound for the entire lockfile, but individual forks can have stricter bounds, e.g. say overall is `>=3.7`, but there's one fork `>=3.7,<3.8` and one fork `>=3.8`

Reply:
This is emulating [the call to `simplify_python_versions` when constructing the `ResolutionGraph`](https://github.com/astral-sh/uv/blob/ibraheem/canonical-markers/crates/uv-resolver/src/resolution/graph.rs#L165).

I agree that marker construction should always have this information, I would like to avoid simplifying requires-python and extras in a separate step.